src/network/admin_server.rs

```diff
@@ -76,12 +89,52 @@ pub enum AdminServiceError {
 const DB_DESTROY_KEY: &[u8] = b"__destroy_all_databases_on_start__";
 
 impl MyAdminService {
-    pub fn new(db_manager: DbManager, mempool_tx: mpsc::Sender<MempoolMessage>) -> Self {
+    pub fn new(
+        db_manager: DbManager,
+        mempool_tx: mpsc::Sender<MempoolMessage>,
+        shard_stores: HashMap<u32, Stores>,
+        block_store: BlockStore,
+        snapshot_config: storage::db::snapshot::Config,
+        fc_network: FarcasterNetwork,
+    ) -> Self {
         Self {
             db_manager,
             mempool_tx,
+            shard_stores,
+            block_store,
+            snapshot_config,
+            fc_network,
         }
     }
+
+    async fn backup_shard(&self, shard_id: u32, db: Arc<RocksDB>, now: i64) -> Result<(), Status> {
+        // TODO(aditi): Eventually, we should upload a metadata file. For now, just clear all existing snapshots and only keep 1 snapshot per shard
+        clear_snapshots(self.fc_network, &self.snapshot_config, shard_id)
+            .await
+            .map_err(|err| Status::from_error(Box::new(err)))?;
+        let db_location = db.location();
+        let backup_dir = Path::new(&db_location)
+            .parent()
```

Comment:
What's the actual directory structure for the backups? I think we should create an explicit directory called `backup` and place the backup there.

Reply:
Done, moved to `.rocks.backup`